sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -292,12 +298,26 @@ static void _on_resp_timeout(void *arg) {
  *
  * This is used in response to an empty ACK.
  *
+ * The current implementation does not touch the timers, but merely sets the
+ * memo's state to GCOAP_MEMO_WAIT. This approach needs less complex code at
+ * the cost of the remaining `send_limit` timers firing and some memory not
+ * being freed until the actual response arrives.
```

Comment:
What happens if the response never arrives?

Reply:
The same as if no ACK were sent: When the retransmission timer fires but the retransmission counter has reached zero, the application is informed of the failure. By not touching the rest of the mechanism, things stay exactly as it would have been.